crates/agent/src/controlplane.rs

```diff
@@ -294,37 +311,15 @@ impl ControlPlane for PGControlPlane {
         names: BTreeSet<String>,
     ) -> anyhow::Result<tables::LiveCatalog> {
         let names = names.into_iter().collect::<Vec<_>>();
-        let rows = agent_sql::live_specs::fetch_live_specs(self.system_user_id, &names, &self.pool)
-            .await?;
-        let mut live = tables::LiveCatalog::default();
-        for row in rows {
-            // Spec type might be null because we used to set it to null when deleting specs.
-            // For recently deleted specs, it will still be present.
-            let Some(catalog_type) = row.spec_type.map(Into::into) else {
-                continue;
-            };
-            let Some(model_json) = row.spec.as_deref() else {
-                continue;
-            };
-            let built_spec_json = row.built_spec.as_ref().ok_or_else(|| {
-                tracing::warn!(catalog_name = %row.catalog_name, id = %row.id, "got row with spec but not built_spec");
-                anyhow::anyhow!("missing built_spec for {:?}, but spec is non-null", row.catalog_name)
-            })?.deref();
-
-            live.add_spec(
-                catalog_type,
-                &row.catalog_name,
-                row.id.into(),
-                row.data_plane_id.into(),
-                row.last_pub_id.into(),
-                row.last_build_id.into(),
-                model_json,
-                built_spec_json,
-                row.dependency_hash,
-            )
-            .with_context(|| format!("deserializing specs for {:?}", row.catalog_name))?;
-        }
+        let mut live = crate::live_specs::get_live_specs(
+            self.system_user_id,
+            &names,
+            None, // don't filter based on user capability
+            &self.pool,
+        )
+        .await?;
 
+        // TODO: Can we stop adding inferred schemas to live specs?
```

Comment:
Now that we update inlined inferred schemas as part of the publication `Initialize` impl, there's no longer any reason why they _need_ to be included as part of a `tables::LiveCatalog`. Validation no longer needs them for anything, because it only injects a placeholder if needed.

So I'm pretty sure that we could just remove them entirely, but would like to take that up as a separate PR since there's still a number of other places where we use it. The `ControlPlane` trait currently implements `get_inferred_schema(models::Collection)` by delegating to `get_live_specs`, for example.

Reply:
Ah, yes, fully agreed.